std/functional.d

```diff
@@ -1814,13 +1814,42 @@ if (isCallable!(F))
     {
         return fp;
     }
+    else static if (is(F Func == Func*) && is(Func == function) && is(Func Params == __parameters))
+    {
+        return () @trusted
```

Comment:
No check if the input function is ``@safe``, therefore this will need another branch to handle those two cases.

Reply:
I’m not sure whether I can your concern here. But I’ve added a unittests that ensures that a `function() @system` won’t get transformed into `delegate() @safe`.